### ISSUE TITLE
Argument for config file when using `LightningArgumentParser` does not exist anymore

### ISSUE THREAD

**Issue by bwconrad** [Lightning-AI/lightning#16181]
### Bug description

When using `LightningArgumentParser`, the argument to pass a single configuration file for all parsed classes, `--config` (as well as `--print_config`) do not exist anymore. You can still pass configs for the individual classes but the argument for a single config covering all classes is gone. 

From what I can see, this first starts happening at version `1.8.0` and works as intended till version `1.7.7`. I didn't see any mention of this feature being removed in the `1.8.0` changelog. 

### How to reproduce the bug
`main.py`

```python
from argparse import Namespace
import pytorch_lightning as pl
import torch
from pytorch_lightning import LightningModule
try:
    from pytorch_lightning.cli import LightningArgumentParser
except:
    from pytorch_lightning.utilities.cli import LightningArgumentParser
from torch.utils.data import DataLoader, Dataset


class RandomDataset(Dataset):
    def __init__(self, size, length):
        self.len = length
        self.data = torch.randn(length, size)

    def __getitem__(self, index):
        return self.data[index]

    def __len__(self):
        return self.len


class BoringModel(LightningModule):
    def __init__(self, dim: int = 32):
        """Boring Model

        Args:
            dim: Hidden dimension
        """

        super().__init__()
        self.layer = torch.nn.Linear(dim, 2)

    def forward(self, x):
        return self.layer(x)

    def training_step(self, batch, batch_idx):
        loss = self(batch).sum()
        self.log("train_loss", loss)
        return {"loss": loss}

    def configure_optimizers(self):
        return torch.optim.SGD(self.layer.parameters(), lr=0.1)


def run():
    parser = LightningArgumentParser()
    parser.add_lightning_class_args(pl.Trainer, "trainer")
    parser.add_lightning_class_args(BoringModel, "model")
    args = parser.parse_args()

    train_data = DataLoader(RandomDataset(32, 64), batch_size=2)
    model = BoringModel(**args["model"])
    trainer = pl.Trainer.from_argparse_args(args["trainer"])
    trainer.fit(model, train_dataloaders=train_data)


if __name__ == "__main__":
    run()
```

`config.yaml`

```
trainer:
  max_epochs: 10
```

Run: `python main.py --config config.yaml`

### Error messages and logs

```
usage: main.py [-h] [--trainer CONFIG] [--trainer.logger.help CLASS_PATH_OR_NAME] [--trainer.logger LOGGER] [--trainer.enable_checkpointing {true,false}]
               [--trainer.callbacks.help CLASS_PATH_OR_NAME] [--trainer.callbacks CALLBACKS] [--trainer.default_root_dir DEFAULT_ROOT_DIR]
               [--trainer.gradient_clip_val GRADIENT_CLIP_VAL] [--trainer.gradient_clip_algorithm GRADIENT_CLIP_ALGORITHM] [--trainer.num_nodes NUM_NODES]
               [--trainer.num_processes NUM_PROCESSES] [--trainer.devices DEVICES] [--trainer.gpus GPUS] [--trainer.auto_select_gpus {true,false}]
               [--trainer.tpu_cores TPU_CORES] [--trainer.ipus IPUS] [--trainer.enable_progress_bar {true,false}] [--trainer.overfit_batches OVERFIT_BATCHES]
               [--trainer.track_grad_norm TRACK_GRAD_NORM] [--trainer.check_val_every_n_epoch CHECK_VAL_EVERY_N_EPOCH] [--trainer.fast_dev_run FAST_DEV_RUN]
               [--trainer.accumulate_grad_batches ACCUMULATE_GRAD_BATCHES] [--trainer.max_epochs MAX_EPOCHS] [--trainer.min_epochs MIN_EPOCHS]
               [--trainer.max_steps MAX_STEPS] [--trainer.min_steps MIN_STEPS] [--trainer.max_time MAX_TIME] [--trainer.limit_train_batches LIMIT_TRAIN_BATCHES]
               [--trainer.limit_val_batches LIMIT_VAL_BATCHES] [--trainer.limit_test_batches LIMIT_TEST_BATCHES]
               [--trainer.limit_predict_batches LIMIT_PREDICT_BATCHES] [--trainer.val_check_interval VAL_CHECK_INTERVAL]
               [--trainer.log_every_n_steps LOG_EVERY_N_STEPS] [--trainer.accelerator.help CLASS_PATH_OR_NAME] [--trainer.accelerator ACCELERATOR]
               [--trainer.strategy.help CLASS_PATH_OR_NAME] [--trainer.strategy STRATEGY] [--trainer.sync_batchnorm {true,false}] [--trainer.precision PRECISION]
               [--trainer.enable_model_summary {true,false}] [--trainer.num_sanity_val_steps NUM_SANITY_VAL_STEPS]
               [--trainer.resume_from_checkpoint RESUME_FROM_CHECKPOINT] [--trainer.profiler.help CLASS_PATH_OR_NAME] [--trainer.profiler PROFILER]
               [--trainer.benchmark {true,false,null}] [--trainer.deterministic DETERMINISTIC]
               [--trainer.reload_dataloaders_every_n_epochs RELOAD_DATALOADERS_EVERY_N_EPOCHS] [--trainer.auto_lr_find AUTO_LR_FIND]
               [--trainer.replace_sampler_ddp {true,false}] [--trainer.detect_anomaly {true,false}] [--trainer.auto_scale_batch_size AUTO_SCALE_BATCH_SIZE]
               [--trainer.plugins.help CLASS_PATH_OR_NAME] [--trainer.plugins PLUGINS] [--trainer.amp_backend AMP_BACKEND] [--trainer.amp_level AMP_LEVEL]
               [--trainer.move_metrics_to_cpu {true,false}] [--trainer.multiple_trainloader_mode MULTIPLE_TRAINLOADER_MODE] [--trainer.inference_mode {true,false}]
               [--model CONFIG] [--model.dim DIM]
main.py: error: Unrecognized arguments: --config config.yaml
```

### Environment

<details>
  <summary>Current environment</summary>

```
#- Lightning Component (e.g. Trainer, LightningModule, LightningApp, LightningWork, LightningFlow): LightningArgumentParser
#- PyTorch Lightning Version (e.g., 1.5.0): 1.8.6
#- Lightning App Version (e.g., 0.5.2):
#- PyTorch Version (e.g., 1.10): 1.13.1
#- Python version (e.g., 3.9): 3.10.8
#- OS (e.g., Linux): Linux
#- CUDA/cuDNN version: 11.7
#- GPU models and configuration:
#- How you installed Lightning(`conda`, `pip`, source): pip
#- Running environment of LightningApp (e.g. local, cloud):
```

</details>

cc @carmocca @mauvilsa

**Comment by carmocca** [comment]
This was changed in #15048 after the report in #15038. @mauvilsa Do you have any suggestions here? Do you think we could find a different solution to the issue in #15038 or is it expected that users add the `--config` parameter manually?

**Comment by mauvilsa** [comment]
`LightningArgumentParser` was initially only though to be used by `LightningCLI` and not standalone. That is why it didn't even cross my mind that #15048 would be a breaking change. Certainly there is nothing wrong with using `LightningArgumentParser` standalone. @bwconrad just out of curiosity to understand the use case, what are your reasons for not using `LightningCLI`?

It is possible that with https://github.com/omni-us/jsonargparse/pull/225, the original issue #15038 would no longer be a problem, though I am not completely sure. Regardless, I think it wouldn't be a good idea to revert #15048. It make more sense that the config option is not added by the parser init. But more importantly, if we add it back, it would be a second breaking change. Anyone using the existing PL 1.8.* versions could similarly decide the use `LightningArgumentParser` standalone and add a config option to the parser. Only one config option per parser can exist, so if on PL upgrade suddenly a config option is added during parser init, that code will fail when the config option is added later.

My recommendation would be to keep things as is, and people using `LightningArgumentParser` standalone add the config option manually.

@bwconrad to add the config option, you just do:

```python
from jsonargparse import ActionConfigFile
...
parser.add_argument('--config', action=ActionConfigFile)
```

**Comment by bwconrad** [comment]
@mauvilsa Using `ActionConfigFile` does exactly what I was looking for. Thanks!

I use `LightningArgumentParser` instead of `LightningCLI` mostly just as a personal preference. I was having issues using `LightningCLI` when I first tried it out right after it got added to PL and have just used `LightningArgumentParser` since. Don't remember the specifics issues but I'll look into using `LightningCLI` again.
